model/src/main/kotlin/Scope.kt

```diff
@@ -63,7 +63,12 @@ data class Scope(
     /**
      * A comparison function to sort scopes by their name.
      */
-    override fun compareTo(other: Scope) = compareValuesBy(this, other) { it.name }
+    override fun compareTo(other: Scope) =
+        when {
+            name < other.name -> -1
+            name > other.name -> 1
+            else -> if (this == other) 0 else 1 // Return arbitrary inequality.
```

Comment:
Again, this is a performance problem.

Reply:
In this concrete case, I don't think there's a performance problem as the expensive comparison would only happen if scope names were equal, which should never be the case as a project cannot have multiple scopes with the same name.